services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/validation/HankeValidator.kt

```diff
@@ -76,12 +43,54 @@ class HankeValidator : ConstraintValidator<ValidHanke, Hanke> {
             ok = false
         }
 
+        ok = ok && checkHankealueet(hanke, context)
         ok = ok && checkTyomaaTiedot(hanke, context)
-        ok = ok && checkHaitat(hanke, context)
 
         return ok
     }
 
+    private fun checkHankealueet(hanke: Hanke, context: ConstraintValidatorContext): Boolean {
+        for (hankealue in hanke.alueet) {
+            // Must be earlier than some relevant maximum date.
+            // The starting date can be in the past, since sometimes the permission to dig a hole is
+            // applied for after the hole has already been dug.
+            if (
+                hankealue.haittaAlkuPvm == null || hankealue.haittaAlkuPvm!!.isAfter(MAXIMUM_DATE)
+            ) {
+                context
+                    .buildConstraintViolationWithTemplate(HankeError.HAI1002.toString())
+                    .addPropertyNode("haittaAlkuPvm")
```

Comment:
This or the context should probably reflect the fact that the violation is in a hankealue, not the hanke itself.

Reply:
Now using a new error code for it.